### PR TITLE
Reworked PR #739 (multiple tags)

### PR DESCRIPTION
**Pull request by KostyaSha**
reworked #739 

@orzeh please review

This change is [Reviewable](https://reviewable.io/reviews/docker-java/docker-java/835)

### CODE REVIEW

**Comment by KostyaSha** [comment]
Reviewed 7 of 7 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/docker-java/docker-java/835)*

File: src/main/java/com/github/dockerjava/netty/exec/BuildImageCmdExec.java [hunk 1]
```diff
@@ -55,9 +49,13 @@ protected Void execute0(BuildImageCmd command, ResultCallback<BuildResponseItem>
         if (dockerFilePath != null && command.getRemote() == null && !"Dockerfile".equals(dockerFilePath)) {
             webTarget = webTarget.queryParam("dockerfile", dockerFilePath);
         }
-        if (command.getTag() != null) {
-            webTarget = webTarget.queryParam("t", command.getTag());
+
+        if (command.getTags() != null && !command.getTags().isEmpty()) {
+            webTarget = webTarget.queryParam("t", command.getTags());
```
Comment (orzeh) [inline src/main/java/com/github/dockerjava/netty/exec/BuildImageCmdExec.java]: should be:
````java
webTarget = webTarget.queryParamSet("t", command.getTags());
````

File: src/main/java/com/github/dockerjava/netty/exec/BuildImageCmdExec.java [hunk 2]
```diff
@@ -55,9 +49,13 @@ protected Void execute0(BuildImageCmd command, ResultCallback<BuildResponseItem>
...
+        } else if (command.getTags() != null) {
+            webTarget = webTarget.queryParamsSet("t", command.getTags());
```
Comment (orzeh) [inline src/main/java/com/github/dockerjava/netty/exec/BuildImageCmdExec.java]: should be:
````java
webTarget = webTarget.queryParam("t", command.getTag());
````

Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/netty/exec/BuildImageCmdExec.java]: damn git conflict

**Comment by KostyaSha** [comment]
@orzeh please review again

**Comment by orzeh** [comment]
@KostyaSha LGTM :+1:

File: src/main/java/com/github/dockerjava/netty/exec/BuildImageCmdExec.java [hunk 3]
```diff
@@ -55,9 +51,13 @@ protected Void execute0(BuildImageCmd command, ResultCallback<BuildResponseItem>
         if (dockerFilePath != null && command.getRemote() == null && !"Dockerfile".equals(dockerFilePath)) {
             webTarget = webTarget.queryParam("dockerfile", dockerFilePath);
         }
-        if (command.getTag() != null) {
-            webTarget = webTarget.queryParam("t", command.getTag());
+
+        if (command.getTags() != null && !command.getTags().isEmpty()) {
+            webTarget = webTarget.queryParamsSet("t", command.getTags());
+        } else if (isNotBlank(command.getTag())) {
+            webTarget = webTarget.queryParam("t", command.getTags());
```
Comment (KostyaSha) [inline src/main/java/com/github/dockerjava/netty/exec/BuildImageCmdExec.java]: !

Comment (orzeh) [inline src/main/java/com/github/dockerjava/netty/exec/BuildImageCmdExec.java]: O_o

**Pull request merged**